### PR TITLE
4557 new module kaijumergeoutputs + stub Kraken2/Kraken2

### LINKED ISSUE #4557: new module: kaiju/mergeOutputs
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by Joon-Klaps**
## PR checklist

Closes #4557 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/kaiju/mergeoutputs/main.nf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/kaiju:1.10.0--h43eeafb_0':
+        'biocontainers/kaiju:1.10.0--h43eeafb_0' }"
+
+    input:
+    tuple val(meta), path(kaiju), path(kraken)
+    path (db)
+
+    output:
+    tuple val(meta), path("*_merged.tsv"), emit: merged
```
Comment (nvnieuwk) [inline modules/nf-core/kaiju/mergeoutputs/main.nf:15]:
```suggestion
    tuple val(meta), path("*.tsv"), emit: merged
```
The guidelines state that no hardcoded paths should be in the output files to keep the file naming as flexible as possible

File: modules/nf-core/kaiju/mergeoutputs/main.nf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args    = task.ext.args ?: ''
+    def prefix  = task.ext.prefix ?: "${meta.id}"
+    def dbnodes = db ? '-t <(find -L ${db} -name "*nodes.dmp")' : ''
+    """
+    kaiju-mergeOutputs \\
+        -i <(sort -k2,2 ${kaiju}) \\
```
Comment (nvnieuwk) [inline modules/nf-core/kaiju/mergeoutputs/main.nf:27]: Maybe also add the possibility to supply arguments to `sort`?

File: modules/nf-core/kaiju/mergeoutputs/main.nf [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        -j <(sort -k2,2 ${kraken}) \\
+        -o ${prefix}_merged.tsv \\
```
Comment (nvnieuwk) [inline modules/nf-core/kaiju/mergeoutputs/main.nf:29]:
```suggestion
        -o ${prefix}.tsv \\
```

File: modules/nf-core/kaiju/mergeoutputs/main.nf [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    "${task.process}":
+        kaiju: \$(echo \$( kaiju -h 2>&1 | sed -n 1p | sed 's/^.*Kaiju //' ))
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}_merged.tsv
```
Comment (nvnieuwk) [inline modules/nf-core/kaiju/mergeoutputs/main.nf:44]:
```suggestion
    touch ${prefix}.tsv
```

File: modules/nf-core/kaiju/mergeoutputs/main.nf (hunk 1 above)
Comment (nvnieuwk) [reply modules/nf-core/kaiju/mergeoutputs/main.nf:15]: Be sure that you check that the input and output file have different names (https://nf-co.re/docs/contributing/modules#naming-conventions part 6)

File: modules/nf-core/kaiju/mergeoutputs/main.nf (hunk 2 above)
Comment (Joon-Klaps) [reply modules/nf-core/kaiju/mergeoutputs/main.nf:27]: Kaiju mergeoutputs only works if it's sorted based on the read ID. Providing args here will provide very little but the module can then easily be broken. So I'm not certain if it's necessary.

Comment (nvnieuwk) [reply modules/nf-core/kaiju/mergeoutputs/main.nf:30]: Okay good point, ignore me here :wink:

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Pull request merged**